Assets/VRTK/Source/Scripts/Utilities/VRTK_SharedMethods.cs

```diff
@@ -472,6 +469,109 @@ public static Vector3 AxisDirection(int axisIndex, Transform givenTransform = nu
+        /// <returns>The found value for the given key in the given dictionary, or the default value if no key is found.</returns>
+        public static TValue GetDictionaryValue<TKey, TValue>(Dictionary<TKey, TValue> dictionary, TKey key, bool setMissingKey = false, TValue setValue = default(TValue))
+        {
+            if (dictionary == null)
+            {
+                return default(TValue);
+            }
+
+            TValue outputValue;
+            if (!dictionary.TryGetValue(key, out outputValue))
+            {
+                if (setMissingKey)
+                {
+                    dictionary.Add(key, setValue);
+                }
+                outputValue = setValue;
+            }
+            return outputValue;
+        }
+
+        /// <summary>
+        /// The AddDictionaryValue method attempts to add a value for the given key in the given dictionary if the key does not already exist. `forceAdd = true` will always set the value even if they key exists.
+        /// </summary>
+        /// <typeparam name="TKey">The datatype for the dictionary key.</typeparam>
+        /// <typeparam name="TValue">The datatype for the dictionary value.</typeparam>
+        /// <param name="dictionary">The dictionary to set the value for.</param>
+        /// <param name="key">The key to set the value for.</param>
+        /// <param name="value">The value to set at the given key in the given dictionary.</param>
+        /// <param name="forceAdd">If this is `true` then the value for the given key will always be set to the provided value. If this is `false` then the value for the given key will only be set if the given key is not found in the given dictionary.</param>
+        public static void AddDictionaryValue<TKey, TValue>(ref Dictionary<TKey, TValue> dictionary, TKey key, TValue value, bool forceAdd = false)
+        {
+            if (forceAdd)
+            {
+                dictionary[key] = value;
+            }
+            else
+            {
+                GetDictionaryValue(dictionary, key, true, value);
+            }
+        }
+
+        /// <summary>
+        /// The RemoveDictionaryKey method attempts to remove an entry from the given dictionary based on the given key.
+        /// </summary>
+        /// <typeparam name="TKey">The datatype for the dictionary key.</typeparam>
+        /// <typeparam name="TValue">The datatype for the dictionary value.</typeparam>
+        /// <param name="dictionary">The dictionary to remove the entry from.</param>
+        /// <param name="key">The key to remove the entry for.</param>
+        /// <returns>Returns `true` if the entry for the given key was removed successfully, returns `false` if the entry for the given key was not found in the dictionary.</returns>
+        public static bool RemoveDictionaryKey<TKey, TValue>(ref Dictionary<TKey, TValue> dictionary, TKey key)
+        {
+            if (dictionary.ContainsKey(key))
```

Comment:
Remove this and change the whole method to
```c#
return dictionary.Remove(key);
```
instead. Or better: Remove this helper method altogether since it's now adding nothing to the existing method on `Dictionary<TValue>`.

Reply:
if the key doesn't exist then won't that throw an exception?

Reply:
Nope, it will instead return true/false like you do:
https://msdn.microsoft.com/en-us/library/kabs04ac(v=vs.90).aspx

Reply:
Ok cool, so this method can be removed then in place of `dictionary.Remove(key)`